src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppCertificate.cs

```diff
@@ -0,0 +1,121 @@
+        [ResourceNameCompleter("Microsoft.Web/sites", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string WebAppName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 2, Mandatory = false, HelpMessage = "The name of the web app slot.")]
+        [ResourceNameCompleter("Microsoft.Web/sites/slots", "ResourceGroupName", "WebAppName")]
+        [ValidateNotNullOrEmpty]
+        public string Slot { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 3, Mandatory = true, HelpMessage = "Custom hostnames associated with web app/slot.")]
+        [ValidateNotNullOrEmpty]
+        public string HostName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Mandatory = false, HelpMessage = "To add the created certificate to WebApp/slot.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter AddBinding { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Mandatory = false, HelpMessage = "Ssl state option. Use either 'SniEnabled' or 'IpBasedEnabled'. Default option is 'SniEnabled'.")]
+        [ValidateNotNullOrEmpty]
+        public SslState? SslState { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrWhiteSpace(ResourceGroupName) && !string.IsNullOrWhiteSpace(WebAppName))
+            {
+                //PSSite webApp = null;
+                var webApp = new PSSite(WebsitesClient.GetWebApp(ResourceGroupName, WebAppName, Slot));
+                var location = webApp.Location;
+
+                Certificate createdCertdetails = null;
+
+                var certificate = new Certificate(
+                    webApp.Location,
+                    canonicalName: HostName,
+                    password: "",
+                    serverFarmId: webApp.ServerFarmId);
+                if (this.ShouldProcess(this.WebAppName, string.Format($"Creating an App service managed certificate for Web App '{WebAppName}'")))
+                {
+                    try
+                    {
+                        WebsitesClient.CreateCertificate(ResourceGroupName, HostName, certificate);
+                    }
+                    catch (DefaultErrorResponseException e)
+                    {
+                        // 'Conflict' exception is thrown when certificate already exists. Let's swallow it and continue.
+                        //'Accepted' exception is thrown by default for create cert method. 
+                        if (e.Response.StatusCode != HttpStatusCode.Conflict &&
+                            e.Response.StatusCode != HttpStatusCode.Accepted)
+                        {
+                            throw;
+                        }
+                    }
+                    //Delay the execution by 5 seconds so that the Certificate will created post it return 'Accepted' status
+                    Thread.Sleep(15000);
```

Comment:
Why do we need the sleep here? we should avoid Sleep in code execution. Is this for handle 202 response?

Reply:
yes sisira this is to handle 202 response.